### DISCUSSION TITLE
Can an xdrip collector share to Dexcom servers for follow?

### DISCUSSION THREAD

**Discussion by corneliusroemer** [NightscoutFoundation/xDrip#3139] [Q&A] [answered]
This is not a feature request but a question whether a certain feature exists:

I've managed to set up xdrip to act as collector for my dexcom G6. I would like to know whether it is possible for xdrip to send the glucose values it collected to the dexcom share servers.

I know that I could set up nightscout, my question is simply whether this feature exists or not.

**Comment by Navid200** [comment, marked as answer]
Yes, this capability exists.  You can find it under `Settings` &#8722;> `Cloud Upload` &#8722;> `Dexcom share server upload`
However, like WearOS, we don't have an active developer taking care of it.
So, if it works, great; if it doesn't, you can ask for help and wait.

Everyone is welcome to join and  contribute.

**Reply by rudysroost** [reply]
> Yes, this capability exists. You can find it under `Settings` −> `Cloud Upload` −> `Dexcom share server upload`. However, like WearOS, we don't have an active developer taking care of it. So, if it works, great; if it doesn't, you can ask for help and wait.
> 
@corneliusroemer It works great for me; I've been using it for years with no issues.

**Reply by corneliusroemer** [reply]
Thanks I found it now in settings:
`Settings>Data Sync/Cloud Upload>Dexcom Share Server Upload`

I've tried to enter my details but I'm not sure it's working. The event log doesn't say anything about a successful upload. @rudysroost do you know how you know it works?

Unfortunately, the Dexcom follow app (not xdrip) seems to have a bug so one can't set up new followers at the moment, something about temporary server error (but google reviews show this issue has been around for months).

**Reply by rudysroost** [reply]
@corneliusroemer The only way I know it works is because my wife uses the Dexcom follow app and gets all of my data. Also, if I have to use a different collector when xDrip fails to collect, I can get the data into xDrip by using Dexcom follow as a data source.

**Reply by corneliusroemer** [reply]
Cool! Do you by any chance know how to reset the password for the following account? I can't set up a new one so I'm trying to get download using my follower's credentials but they've forgotten their password.

**Reply by corneliusroemer** [reply]
@rudysroost what do you input as the serial number? I have no clue what to provide. I might have to dive into the code some day to see how to improve the logging.

**Comment by corneliusroemer** [comment]
This is the code for the sharing service in case anyone wants to look at it - I'll try to debug this with curl http requests
https://github.com/NightscoutFoundation/xDrip/blob/330f63b615f69e647e7a748ba8575294fbc86520/app/src/main/java/com/eveningoutpost/dexdrip/sharemodels/ShareRest.java#L44

Good documentation for fixing things is here: 
- https://github.com/TooTallNate/dexcom-share/blob/master/src/index.ts
- https://github.com/gagebenne/pydexcom
- https://gist.github.com/StephenBlackWasAlreadyTaken/adb0525344bedade1e25
- https://github.com/nightscout/share2nightscout-bridge
